source/common/upstream/eds.cc

```diff
@@ -117,9 +117,15 @@ void EdsClusterImpl::onConfigUpdate(const Protobuf::RepeatedPtrField<ProtobufWkt
   if (!validateUpdateSize(resources.size())) {
     return;
   }
-  auto cluster_load_assignment =
-      MessageUtil::anyConvertAndValidate<envoy::config::endpoint::v3::ClusterLoadAssignment>(
-          resources[0], validation_visitor_);
+  envoy::config::endpoint::v3::ClusterLoadAssignment cluster_load_assignment;
+  if (validation_visitor_.skipValidation()) {
+    cluster_load_assignment =
+        MessageUtil::anyConvert<envoy::config::endpoint::v3::ClusterLoadAssignment>(resources[0]);
+  } else {
+    cluster_load_assignment =
+        MessageUtil::anyConvertAndValidate<envoy::config::endpoint::v3::ClusterLoadAssignment>(
+            resources[0], validation_visitor_);
```

Comment:
Can we move this `skipValidation()` conditioning inside `MessageUtil::anyConvertAndValidate()`? This seems like something to do generically, rather than on a per-resource type basis, i.e. just for EDS here.

Reply:
Done.